avocado/utils/software_manager.py

```diff
@@ -885,15 +885,25 @@ def add_repo(self, repo):
         :param repo: Repository string. Example:
                 'deb http://archive.ubuntu.com/ubuntu/ maverick universe'
         """
-        with open(self.repo_file_path, 'r') as repo_file:
-            repo_file_contents = repo_file.read()
-            if repo not in repo_file_contents:
-                try:
-                    add_cmd = "echo '%s' > %s" % (repo, self.repo_file_path)
-                    process.system(add_cmd, shell=True, sudo=True)
-                    return True
-                except process.CmdError:
-                    return False
+        def _add_repo_file():
+            add_cmd = "bash -c \"echo '%s' > %s\"" % (repo, self.repo_file_path)
```

Comment:
I can't see why `bash -c` would be needed here in addition to `shell=True`.  Care to share the reason?

Reply:
Due to the way shell handles redirection - Simply executing `sudo echo 'foo' > /protected/file` doesn't work, since it's trying to access the `/protected/file` with non escalated user's permissions. Executing the entire command on a shell started with sudo is the way to make this work properly.

Reply:
OK, makes sense.  One last point, though: since you provided some compatibility fixes for MacOS, is bash universally available there? Or should we be using `sh` instead?

Reply:
Bash is the default macOS shell, fortunately. So yeah, we can use that safely, not to mention that software_manager is eminently a Linux library.

Reply:
One can argue that you can use `echo %s | sudo tee %s` (I'm used to use this) but those who allow to sudo `tee` probably also allow `echo` as well 😄